examples/acados_template/python/pendulum_example/generate_c_code.py

```diff
@@ -98,6 +116,20 @@
 Vx_e[3,3] = 1.0
 
 nlp_cost.Vx_e = Vx_e
+if FORMULATION == 'NLS':
+    x = SX.sym('x', 4, 1)
+    u = SX.sym('u', 1, 1)
+    ocp.cost_r.expr = vertcat(u, x) 
```

Comment:
Note that this is called `cost_expr_y`, which might be more consistent with the LS cost type, since the residual is `y - y_ref `.

Reply:
I agree, I am planning to move this auxiliary class into acados.cost anyway, stay tuned :P

Reply:
Agree with Jonathan. The advantage of having `expr_y - y_ref` instead of `r` is that `y_ref` is numerical and therefore it can be changed easily when there are changes in the reference (which is a common case). The use of `r` requires the use of parameters instead.

If you agree with that, I guess it's easier to code `expr_y` from the beginning, instead of coding first as `r` and having to change it later on ;)

Reply:
The formulation is already r - y_ref. I guess I can just change r to expr_y ;)